### PR TITLE
use RepeatedPtrField<tipb::Expr> instead of vector<tipb::Expr *>

### PR DESCRIPTION
**Pull request by gengliqi**
### What problem does this PR solve?

Issue Number: ref https://github.com/pingcap/tiflash/issues/6233

Problem Summary:
Use `RepeatedPtrField<tipb::Expr>` instead of `vector<tipb::Expr *>` to reduce useless copy and make code clear.

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [x] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
None
```

### CODE REVIEW

**Review by SeaRise — APPROVED** [review]

**Review by ywqzzy — APPROVED** [review]

File: dbms/src/Storages/DeltaMerge/FilterParser/FilterParser.cpp [hunk 1]
```diff
@@ -398,15 +398,15 @@ RSOperatorPtr FilterParser::parseDAGQuery(const DAGQueryInfo & dag_info,
...
     {
-        op = cop::tryParse(*dag_info.filters[0], columns_to_read, creator, dag_info.timezone_info, log);
+        op = cop::tryParse(dag_info.filters[0], columns_to_read, creator, dag_info.timezone_info, log);
     }
     else
     {
         /// By default, multiple conditions with operator "and"
         RSOperators children;
-        for (size_t i = 0; i < dag_info.filters.size(); ++i)
+        for (int i = 0; i < dag_info.filters.size(); ++i)
```
Comment (LittleFall) [inline dbms/src/Storages/DeltaMerge/FilterParser/FilterParser.cpp:407]: why change this?
```suggestion
        for (size_t i = 0; i < dag_info.filters.size(); ++i)
```

Comment (gengliqi) [reply dbms/src/Storages/DeltaMerge/FilterParser/FilterParser.cpp:407]: You can see the generated code from protobuf. The return value of `RepeatedPtrField::size` is `int`.

Comment (LittleFall) [reply dbms/src/Storages/DeltaMerge/FilterParser/FilterParser.cpp:407]: indeed

**Pull request merged**